# 📝 Practical Exercise – Apple Stocks Portfolio

**Scenario:**
John is an investor with Apple stocks.
- Each Apple stock is currently valued at **$135**.
- He  bought **20 stocks at $60 each**, then acquired an additional **10 at $145 each**.
- He  sold **8  stocks at $120 each**.

Now, John is evaluating the financial merits of selling his remaining stocks and is curious about the potential tax implications.

---

### Tasks:
1. Calculate and display the total value of the current Apple stocks in John’s portfolio.
2. Calculate and display the average purchase price of John’s Apple stocks.
3. Calculate and display the potential total capital gain if John sells all his Apple stocks now.
4. John holds dual citizenship in **France** and **Belgium**.
   - In France: capital gains tax = **30%**.
   - In Belgium: capital gains are **tax-exempt up to $500**, then **40% applies** beyond this threshold.

   Calculate and display the taxes John would incur on his capital gains in France and Belgium.


In [6]:
stock_value_now = 135  # current stock value
stocks_bought_first = 20
price_first = 60  # price at which the first set of stocks were bought
stocks_bought_second = 10
price_second = 145  # price at which the second set of stocks were bought
stocks_sold = 8
price_sold = 120  # price at which stocks were sold
France_tax_rate, Belgium_tax_rate = 0.30 , 0.40
tax_exempt_threshold = 500

In [7]:
#Calculate and display the total value of the current Apple stocks in John’s portfolio
total_stock = stocks_bought_first + stocks_bought_second - stocks_sold
total_value = total_stock * stock_value_now
print(f"Total value of the current Apple stocks in John's portfolio: ${total_value:.2f}")

Total value of the current Apple stocks in John's portfolio: $2970.00


In [8]:
#Calculate and display the average purchase price of John’s Apple stocks
average_purchase_price = (stocks_bought_first * price_first) + (stocks_bought_second * price_second) / (stocks_bought_first + stocks_bought_second)
print(f"Average purchase price of John's Apple stocks: ${average_purchase_price:.2f}")

Average purchase price of John's Apple stocks: $1248.33


In [9]:
#Calculate and display the potential total capital gain if John sells all his Apple stocks now
potential_total_capital_gain = total_value - (stocks_bought_first * price_first) - (stocks_bought_second * price_second)
print(f"Potential total capital gain: ${potential_total_capital_gain:.2f}")

Potential total capital gain: $320.00


In [10]:
#John holds dual citizenship in France and Belgium. In France, the tax rate on capital gains is 30%. In Belgium, capital gains are tax-exempt up to $500, and a 40% tax rate applies beyond this threshold. Calculate and display the taxes John would incur on his capital gains in France and Belgium
tax_in_France = potential_total_capital_gain * France_tax_rate
tax_belgium = max(0, (potential_total_capital_gain - tax_exempt_threshold) * Belgium_tax_rate)
print(f"Taxes in France: ${tax_in_France:.2f}")
print(f"Taxes in Belgium: ${tax_belgium:.2f}")

Taxes in France: $96.00
Taxes in Belgium: $0.00


## ✅ Results:
- Total value: $2970.00
- Average purchase price: $1248.33
- Capital gain: $320.00
- France tax: $96.00
- Belgium tax: $0.00
